app/src/main/java/com/hedvig/app/feature/embark/passages/multiaction/MultiActionViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.hedvig.app.feature.embark.passages.multiaction
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import com.hedvig.app.feature.embark.passages.multiaction.MultiActionItem.*
+import com.hedvig.app.util.LiveEvent
+import com.hedvig.app.util.extensions.replace
+
+class MultiActionViewModel(
+    private val multiActionParams: MultiActionParams
+) : ViewModel() {
+
+    private val _addedComponents = MutableLiveData<List<Component>>(listOf())
+    val components: LiveData<List<MultiActionItem>> = Transformations.map(_addedComponents) { components ->
+        if (components.size < multiActionParams.maxAmount) {
+            val addButton = AddButton(::createNewComponent)
+            listOf(addButton) + components
+        } else {
+            components
+        }
+    }
```

Comment:
Using the `LiveData.map`-extension reads a little better, I think. We can also omit the explicit type and let it be inferred, if we want 🙂 

```suggestion
    val components = _addedComponents.map { components ->
        if (components.size < multiActionParams.maxAmount) {
            val addButton = AddButton(::createNewComponent)
            listOf(addButton) + components
        } else {
            components
        }
    }
```